### PR TITLE
33 35 segment damage income calculation and sunk ship segment bonuses

### LINKED ISSUE #33: [implementation] sunk ship bonus
Implement bonus when sinking ship

### LINKED ISSUE #35: [bug] income adjustment on damaged ships
Should properly set income when ship is damaged

### PR DESCRIPTION
**Pull request by Andrew1431**
Fixes #33, Fixes #35

### CODE REVIEW

File: src/reducers/game/gameSlice.ts [hunk 1]
```diff
@@ -204,21 +205,26 @@ export const gameSlice = createSlice({
...
         const pillage = settings.upgrades.pillage[state.levels.pillage];
-        let bounty = 0;
+        let bounty = 0,
+          bountySegments = 0;
         action.hits
           .filter((h) => h.userId !== state.users.self.id)
           .forEach((hit) => {
             bounty += pillage.earningsPerSegment;
+            if (hit.sunk) {
+              bountySegments += 1;
```
Comment (ryan1431) [inline src/reducers/game/gameSlice.ts:223]: Bounty segments should be dependent on the level of the pillage upgrade

File: src/reducers/game/gameSlice.ts [hunk 2]
```diff
@@ -235,14 +241,19 @@ export const gameSlice = createSlice({
...
+        let bounty = 0,
+          bountySegments = 0;
         const pillage = settings.upgrades.pillage[action.pillage || 0];
         action.hits
           .filter((h) => h.userId !== state.users.self.id)
-          .forEach(() => {
+          .forEach((h) => {
             bounty += pillage.earningsPerSegment;
+            if (h.sunk) {
+              bountySegments += 1;
```
Comment (ryan1431) [inline src/reducers/game/gameSlice.ts:252]: Same as ^

**Review by ryan1431 — CHANGES_REQUESTED** [review]

File: src/reducers/game/gameSlice.ts (hunk 1 above)
Comment (Andrew1431) [reply src/reducers/game/gameSlice.ts]: boss catch

**Review by ryan1431 — APPROVED** [review]
Good work!

**Pull request merged**